client/my-sites/hosting/site-wp-admin-card/use-select-interface-mutation.ts

```diff
@@ -0,0 +1,36 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+import { useDispatch } from 'calypso/state';
+import { requestSite } from 'calypso/state/sites/actions';
+
+const TOGGLE_SITE_INTERFACE_MUTATION_KEY = 'set-site-interface-mutation-key';
+
+export const useSiteInterfaceMutation = ( siteId: number ) => {
+	const queryClient = useQueryClient();
+	const dispatch = useDispatch();
+	const queryKey = [ TOGGLE_SITE_INTERFACE_MUTATION_KEY, siteId ];
+	const mutation = useMutation( {
+		mutationFn: async ( enabled: boolean ) => {
+			const selectedInterface = enabled ? 'wp-admin' : 'calypso';
+			return wp.req.post( {
+				path: `/sites/${ siteId }/hosting/admin-interface?interface=${ selectedInterface }`,
+				apiNamespace: 'wpcom/v2',
+			} );
+		},
+		mutationKey: queryKey,
+		onError( _err, _newActive, prevValue ) {
+			// Revert to previous settings on failure
+			queryClient.setQueryData( queryKey, Boolean( prevValue ) );
+		},
```

Comment:
Not sure if this is doing something 🤔 , because we don't read the data with the `useQuery`  hook.
I tested it without network, but it doesn't fail, because the mutation is triggered only when the connection is restored. Anyways it's a good place to handle the error.

Reply:
You can trigger the error if you make the parameter that goes to API faulty - for instance change:

`const selectedInterface = enabled ? 'wp-admin' : 'calypso';`

to something like `calyps` and try the toggle. It will go work but fetch the previous option. I am planning to re-use to add an error notice when there is a problem with setting the interface.